### DISCUSSION TITLE
Use virtual address remapping technology to compact LOH in GC

### DISCUSSION THREAD

**Discussion by ygc369** [dotnet/runtime#105086] [Ideas]
refer to the idea in https://github.com/dotnet/runtime/issues/4076

**Comment by huoyaoyuan** [comment]
In the view of virtual memory, it should be effectively an more efficient memcpy in virtual memory space.

I still can't find any information about this behavior in Windows/macOS, only for IOMMU but not physical memory.

The GC team may not get prioritized for this. As the original post says, Compacting LOH is not so impactful for 64 bit processes. The current focus is more on optimizing small object heaps. Community member is welcomed to try this, and presumably measure its impact.
